docs/developer-docs/JavaScript-Reference.md

```diff
@@ -1,3 +1,22 @@
+## `wp_listen_for_consent_change`
+
+Event to listen for consent changes.
+
+### Example
+
+```js
+document.addEventListener( 'wp_listen_for_consent_change', function ( e ) {
+  var changedConsentCategory = e.detail;
+  for ( var key in changedConsentCategory ) {
+    if ( changedConsentCategory.hasOwnProperty( key ) ) {
```

Comment:
This example seems quite complex and a bit tricky to read. Wouldn’t `if ( e.detail.marketing && e.detail.marketing === 'allow' )` achieve the same?

Reply:
probably, again this was copy pasta, but can update ours and maybe propose the change upstream as well.

```suggestion
    if ( e.detail.marketing && e.detail.marketing === 'allow' ) {
```

Reply:
You could remove the `for` loop too with this change